front/lib/resources/app_resource.ts

```diff
@@ -126,62 +120,71 @@ export class AppResource extends ResourceWithVault<App> {
     {
       name,
       description,
-      visibility,
     }: {
       name: string;
       description: string | null;
-      visibility: AppVisibility;
     }
   ) {
     assert(this.canWrite(auth), "Unauthorized write attempt");
+
     await this.update({
       name,
       description,
-      visibility,
-    });
-  }
-
-  async markAsDeleted(auth: Authenticator) {
-    assert(this.canWrite(auth), "Unauthorized write attempt");
-    await this.update({
-      visibility: "deleted",
     });
   }
 
   // Deletion.
 
   async delete(auth: Authenticator): Promise<Result<undefined, Error>> {
+    assert(this.canWrite(auth), "Unauthorized write attempt");
+
     try {
-      await frontSequelize.transaction(async (t) => {
-        await RunResource.deleteAllByAppId(this.id, t);
-        await Clone.destroy({
-          where: {
-            [Op.or]: [{ fromId: this.id }, { toId: this.id }],
-          },
-          transaction: t,
-        });
-        const res = await DatasetResource.deleteForApp(auth, this, t);
-        if (res.isErr()) {
-          // Interrupt the transaction if there was an error deleting datasets.
-          throw res.error;
-        }
-        await App.destroy({
-          where: {
-            workspaceId: auth.getNonNullableWorkspace().id,
-            id: this.id,
-          },
-          transaction: t,
-          // Use 'hardDelete: true' to ensure the record is permanently deleted from the database,
-          // bypassing the soft deletion in place.
-          hardDelete: true,
-        });
+      // Soft delete the app.
+      await AppModel.destroy({
+        where: {
+          workspaceId: auth.getNonNullableWorkspace().id,
+          id: this.id,
+        },
       });
+
       return new Ok(undefined);
     } catch (err) {
       return new Err(err as Error);
     }
   }
 
+  async destroy(auth: Authenticator) {
```

Comment:
Same as the other PR let's be more explicit and avoid introducing a new verb. delete(hard: boolean)